### DISCUSSION TITLE
Rewrite it in Rust

### DISCUSSION THREAD

**Discussion by MaksExtraIdiot** [InfiniTimeOrg/InfiniTime#1858] [Ideas]
Rust is a safe new programming language developed by Rust Foundation that prevents data races and memory related vulnerabilities. You should be using Rust instead of unsafe languages like C++. I suggest you begin rewriting InfiniTime in Rust and keep up with modern trends in safety and performance.

**Comment by Avamander** [comment]
There are no plans on rewriting InfiniTime in Rust in any nearby future. There are a few attempts written in Rust and the Tock kernel that could be used, if you're personally willing to put in the work.

**Comment by ildar** [comment]
I agree it's no use of rewriting.
Actually it's a nice idea to build up an *alternative* FW using Rust
picking the multiple "+" of InfiniTime.
The good base for that could be: https://github.com/dbrgn/pinetime-rtic
Plz drop a note here if any work is done in this direction. Thenks.
